src/util/dapps.js

```diff
@@ -208,63 +208,50 @@ export function fetchRegistryApp (api, dappReg, appId) {
       dappReg.getManifest(appId)
     ])
     .then(([imageId, contentId, manifestId]) => {
-      const app = {
-        id: appId,
-        image: IconCache.hashToImage(imageId),
-        contentHash: bytesToHex(contentId).substr(2),
-        manifestHash: bytesToHex(manifestId).substr(2),
-        type: 'network',
-        visible: true
-      };
-
-      return fetchManifest(api, app.manifestHash)
-        .then((manifest) => {
-          if (manifest) {
-            app.manifestHash = null;
-
-            // Add usefull manifest fields to app
-            Object.assign(app, pick(manifest, ['author', 'description', 'name', 'version']));
-          }
-
-          return app;
-        });
-    })
-    .then((app) => {
-      // Keep dapps that has a Manifest File and an Id
-      const dapp = (app.manifestHash || !app.id) ? null : app;
-
-      return dapp;
-    })
-    .catch((error) => {
-      console.warn('DappsStore:fetchRegistryApp', error);
-    });
-}
-
-export function fetchManifest (api, manifestHash) {
-  if (/^(0x)?0+/.test(manifestHash)) {
-    return Promise.resolve(null);
-  }
-
-  return api.parity
-    .dappsUrl()
-    .then(dappsUrl => {
-      const protocol = window.location.protocol === 'https:' ? 'https:' : 'http:';
+      const imageHash = bytesToHex(imageId).substr(2);
+      const contentHash = bytesToHex(contentId).substr(2);
+      const manifestHash = bytesToHex(manifestId).substr(2);
+
+      return Promise.all([
+        HashFetch.get().fetch(api, imageHash, 'file').catch((e) => { console.warn(`Couldn't download icon for dapp ${appId}`, e); }),
```

Comment:
this will resolve the promise to `undefined` in case of error, is that ok?

Reply:
yes, this is expected. we want to still display the dapp if the icon fetch failed. the only consequence is that the icon url will be `file://undefined`